apis/python/src/tiledbsoma/options/_soma_tiledb_context.py

```diff
@@ -146,9 +151,19 @@ def timestamp(self) -> Optional[datetime.datetime]:
             return None
         return ms_to_datetime(self.timestamp_ms)
 
+    @property
+    def native_context(self) -> clib.SOMAContext:
+        """The C++ SOMAContext for this SOMA context."""
+        with self._lock:
+            if self._native_context is None:
+                self._native_context = clib.SOMAContext(
+                    {k: str(self.tiledb_config[k]) for k in self.tiledb_config}
+                )
+            return self._native_context
```

Comment:
How do you use this lock correctly? Something seems to be holding on to it and preventing anything from running. I just tested locally and Pytest also freezes but commenting this out makes everything run again.

Reply:
the lock is not re-entrant, and this code calls self.tiledb_config which also tries to acquire it.

Solutions:
* unwind the nested locks
* or switch to a re-entrant lock (threading.RLock instead of .Lock)

Reply:
Doc: https://docs.python.org/3/library/threading.html#rlock-objects